packages/client/tests/functional/engine-selection-logic/tests.ts

```diff
@@ -0,0 +1,45 @@
+import { NewPrismaClient } from '../_utils/types'
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let newPrismaClient: NewPrismaClient<typeof PrismaClient>
+
+testMatrix.setupTestSuite(
+  (suiteConfig, suiteMeta, clientMeta) => {
+    const OLD_ENV = process.env
+
+    beforeEach(() => {
+      process.env = { ...OLD_ENV } // Make a copy
+    })
+
+    afterAll(() => {
+      process.env = OLD_ENV // Restore old environment
+    })
+
+    testIf(clientMeta.dataProxy /** = --no-engine */)('--no-engine prevents from using the other engines', async () => {
+      process.env[`DATABASE_URI_${suiteConfig.provider}`] = 'postgresql://postgres:password@localhost:5432/db'
+
+      const prisma = newPrismaClient()
+      const promise = prisma.$connect()
+
+      // proof that the correct engine is used
+      await expect(promise).rejects.toThrowErrorMatchingInlineSnapshot(`Datasource URL must use prisma:// protocol`)
```

Comment:
The error shows that when using `--no-engine` (formerly, `--data-proxy`), you can only use the Data proxy.